client/src/main/java/com/paypal/selion/platform/grid/LocalSelendroidNode.java

```diff
@@ -15,191 +15,168 @@
 
 package com.paypal.selion.platform.grid;
 
-import io.selendroid.grid.SelendroidSessionProxy;
-import io.selendroid.standalone.SelendroidConfiguration;
-import io.selendroid.standalone.SelendroidLauncher;
-
-import java.util.ArrayList;
-import java.util.List;
-import java.util.logging.Handler;
-import java.util.logging.Level;
-import java.util.logging.Logger;
-
 import org.apache.commons.configuration.ConversionException;
-import org.openqa.grid.common.exception.GridException;
+import org.apache.commons.lang.StringUtils;
+import org.openqa.selenium.net.NetworkUtils;
+import org.openqa.selenium.net.PortProber;
 
+import com.google.common.annotations.Beta;
 import com.paypal.selion.configuration.Config;
 import com.paypal.selion.configuration.Config.ConfigProperty;
+import com.paypal.selion.grid.ProcessLauncherOptions;
+import com.paypal.selion.grid.SelendroidJarSpawner;
 import com.paypal.selion.logger.SeLionLogger;
 import com.paypal.test.utilities.logging.SimpleLogger;
 
 /**
- * A singleton that is responsible for encapsulating all the logic w.r.t starting/shutting down a local android driver
- * node.
- * 
+ * A singleton that is responsible for encapsulating all the logic w.r.t starting/shutting down a local selendroid node.
  */
-class LocalSelendroidNode extends AbstractNode implements LocalServerComponent {
+@Beta
+final class LocalSelendroidNode extends AbstractBaseLocalServerComponent implements LocalServerComponent {
+    private static final SimpleLogger LOGGER = SeLionLogger.getLogger();
+    private static volatile LocalSelendroidNode instance;
+
+    static synchronized final LocalServerComponent getSingleton() {
```

Comment:
Why do we need an explicit method such as this one to check if the node/hub is up and running ? In any case all we are doing is just spinning off a sub process and there is no assurance that a node was really up and running due to us spinning it off as a sub process until and unless we can actually access that standalone JVM and interact with it. So what are we asserting here ?
Shouldn't we just be resorting to invoking the Grid API calls i.e.,
* `grid/api/hub/` to get the Hub status
* `/grid/api/proxy/` to get the node status
* `/wd/hub/status` to get the Selendroid node status.

Once we take this route then we don't need a `getSingleton()` implementation in the base class nor the child classes.